src/data-table/data-table.js

```diff
@@ -76,6 +76,10 @@
           row.classList.add(this.CssClasses_.IS_SELECTED);
         } else {
           row.classList.remove(this.CssClasses_.IS_SELECTED);
+          var thead_checkbox = this.element_.tHead.querySelector('.mdl-checkbox.is-checked');
+          if(thead_checkbox !== null){
```

Comment:
This if check should also check if the checkbox is checked before doing the unchecking. Just to avoid extra routines happening when it's known they aren't needed.